pkg/clustermesh/mcsapi/service_controller.go

```diff
@@ -236,6 +236,12 @@ func (r *mcsAPIServiceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	if localSvc != nil {
 		svc.Spec.Selector = localSvc.Spec.Selector
 		svc.Spec.Ports = localSvc.Spec.Ports
+
+		// Use the local Service on creation to save up a potential round trip of
```

Comment:
I don't quite understand the relationship with the comment and the code. How does setting the ClusterIP save a "roundtrip"? What was the original reason for adding this?

Reply:
So this is triggered when the `ServiceImport` object is not yet created and you only have a `ServiceExport`.

Previously in that case the Service was always created as non headless which means that if the `ServiceImport` that would most likely be created right after would retrigger a reconcile of this controller and if it headless the "derived service" then be deleted and recreated.

With this new condition we rely on the local service to determine the headlessness of the derived service so that we might save up the deletion+recreation/not have to switch the headlessness. This however won't be the case if there is a conflict on this property though.

I will updated the comment to be a bit more explicit there.